### ISSUE TITLE
Improve documentation with Literate programming

### ISSUE THREAD

**Issue by jmmercadom** [mavcunha/devcli#1]
**Problem**

It requires a lot of effort to know how to use the devcli (read the code, read the examples). Even though the source code comes with examples of subcommands for connecting with a vpn, vagrant, or share git repositories; it is hard to differentiate what is part of the core functionality of devcli and what is merely part of the solution for the example.

**Possible solutions**

There are plenty things that can be done to improve documentation. Here are some from the top of my head:

* Add a tutorial.md explaining step by step how to customize devcli;
* Move the examples to another repository to avoid confusions;
* Add more comments in the source.

I would like to suggest the last options as a viable solution to describe what things are important (part of the core) and what is part of a example.

Basically my vision is to create a documentation pretty similar as what jasminejs has https://jasmine.github.io/2.4/introduction.html, it is called literate programming and is meant to use the comments from the source code to generate the documentation https://en.wikipedia.org/wiki/Literate_programming.

I would like to know your comments, ideas. If we agree in some solution then later we can figure out "the how", by now I am more interested in "the what" (improve documentation).

**Comment by mavcunha** [comment]
Hi @jmmercadom, 
 Sorry for the long delay and thanks for the note. I do agree that we do not have yet a good documentation and it can be challenging for folks that are not familiar with Bash idiosyncrasies. Worst, the design decisions are not clear as you put well. 
 
I'm familiar with literate programming but I don't think trying to do it in bash would be easy nor practical, at least not at the cweb levels I remember. A couple of documentation pages should suffice for most people IMO. The tool is pretty stable (at least for my daily needs) for now so the documentation won't be outdated soon.
